src/extension/gogoduck/src/test/java/au/org/emii/gogoduck/worker/GoGoDuckModuleTest.java

```diff
@@ -0,0 +1,72 @@
+package au.org.emii.gogoduck.worker;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import au.org.emii.gogoduck.worker.GoGoDuck.UserLog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ucar.nc2.Attribute;
+import ucar.nc2.NetcdfFile;
+import ucar.nc2.NetcdfFileWriter;
+
+public class GoGoDuckModuleTest {
+    public GoGoDuckModule ggdm = null;
+
+    @Before
+    public void beforeEach() {
+        ggdm = new GoGoDuckModule();
+        ggdm.init("", "", "TIME,1,2;LONGITUDE,2,3", null);
+    }
+
+    @Test
+    public void testGetSubsetParameters() throws Exception {
+        // Make sure it removes `TIME` subset parameter
+        assertFalse(ggdm.getSubsetParameters().containsKey("TIME"));
+
+        // But did that only on a copy of the subset parameters
+        Field privateSubset = GoGoDuckModule.class.getDeclaredField("subset");
+        privateSubset.setAccessible(true);
+        SubsetParameters sp = (SubsetParameters) privateSubset.get(ggdm);
+        assertTrue(sp.containsKey("TIME"));
+    }
+
+    public class GoGoDuckModule_test extends GoGoDuckModule {
+        @Override
+        protected List<Attribute> getGlobalAttributesToUpdate(NetcdfFile nc) {
+            List<Attribute> newAttributeList = new ArrayList<Attribute>();
+            newAttributeList.add(new Attribute("test_attribute", "test_value"));
+            return newAttributeList;
+        }
+    }
+
+    @Test
+    public void testUpdateMetadata() throws Exception {
+        File sampleNetcdf = new File("src/test/resources/file.nc").getAbsoluteFile();
```

Comment:
Looking at this test has made me realise that many java APIs tend to deal with streams, rather than files, for added flexibility (and probably performance).  So here for example, you could avoid dealing with real files, and use say `ByteArrayOutputStream` - faster (important for unit tests), and cleaner, because you don't have to cleanup after yourself.